python/kfserving/kfserving/test_storage.py

```diff
@@ -34,3 +40,39 @@ def test_no_prefix_local_path():
     relative_path = '.'
     assert kfserving.Storage.download(abs_path) == abs_path
     assert kfserving.Storage.download(relative_path) == relative_path
+
+
+@mock.patch(STORAGE_MODULE + '.storage')
+def test_mock_gcs(mock_storage):
+    gcs_path = 'gs://foo/bar'
+    mock_obj = mock.MagicMock()
+    mock_obj.name = 'mock.object'
+    mock_storage.Client().bucket().list_blobs().__iter__.return_value = [mock_obj]
+    assert kfserving.Storage.download(gcs_path)
+
+
+@mock.patch('urllib3.PoolManager')
+@mock.patch(STORAGE_MODULE + '.Minio')
+def test_mock_minio(mock_connection, mock_minio):
+    minio_path = 's3://foo/bar'
+    # Create mock connection
+    mock_server = mock.MagicMock()
+    mock_connection.return_value = mock_server
+    # Create mock client
+    mock_minio.return_value = Minio("s3.us.cloud-object-storage.appdomain.cloud", secure=True)
+    mock_obj = mock.MagicMock()
+    mock_obj.object_name = 'mock.object'
+    mock_minio.list_objects().__iter__.return_value = [mock_obj]
+    assert kfserving.Storage.download(minio_path)
+
+
+@mock.patch(STORAGE_MODULE + '.Minio')
+def test_no_permission_buckets(mock_minio):
+    bad_s3_path = "s3://random/path"
+    bad_gcs_path = "gs://random/path"
+    # Access private buckets without credentials
+    mock_minio.return_value = Minio("s3.us.cloud-object-storage.appdomain.cloud", secure=True)
```

Comment:
I think here we need to mock `urllib3.PoolManager` as well to mock the exception response
`mock_connection.side_effect = error.AccessDenied(None)`

Reply:
done, thanks for the advice.